### ISSUE TITLE
$tickets = $client->tickets()->findAll();

### ISSUE THREAD

**Issue by aprast** [zendesk/zendesk_api_client_php#194]
Very simply not problem with autoload or authentication.

Just trying to get all tickets:

$tickets = $client->tickets()->findAll();
print_r($tickets);

Produces:
PHP Fatal error:  Call to a member function getReasonPhrase() on a non-object in /var/www/html/ZD/vendor/zendesk/zendesk_api_client_php/src/Zendesk/API/Exceptions/ApiResponseException.php on line 22

I tried to follow some of the previous reports of similar challenges and I wan unable to find an answer.

Can you provide some insight into what I am doing incorrectly?

**Comment by jwswj** [comment]
Can you provide more context on what you're doing before making this call? Can you get the the example (https://github.com/zendesk/zendesk_api_client_php/blob/master/samples/getTickets.php) to work?

**Comment by aprast** [comment]
I took the example as is with my credentials:

Here is what I get:

PHP Fatal error:  Call to a member function getReasonPhrase() on a
non-object in
/var/www/html/ZD/vendor/zendesk/zendesk_api_client_php/src/Zendesk/API/Exceptions/ApiResponseException.php
on line 22

Here is my composer.json file.

{
    "require": {
        "zendesk/zendesk_api_client_php": "2.x-dev"
    }
}

**Comment by jwswj** [comment]
@aprast you'd going to have to give me some more details. Where are you running this script from?

**Comment by aprast** [comment]
You took twelve days to respond to me and you close the ticket after 3 days?

Seriously?

Does your module even work?

I am on rackspace and install via composer.  I try to run an example as is
with a different api key and it does not work?

My fault?

Come on, you are better than that.

Just help me get the basics running.

I can run on a mac, ubuntu, etc.  just create a clean directory and try
it.  It does not work.  I had to write my own code just calling the api's
directly via curl...

**Comment by jwswj** [comment]
Here is an example of the client being installed on a newly provisioned Digital Ocean Ubuntu 14.04 x64 instance: [Video recording](https://www.ilosvideos.com/vv/?v=CsDzh5E8hRmm)

I install php, composer, then pull tickets from our API using the API client. These are the commands I run in the video:

```bash
# Install PHP
sudo apt-get install php5 libapache2-mod-php5 php5-mcrypt

# Install Composer
curl -sS https://getcomposer.org/installer | php
mv composer.phar /usr/local/bin/composer

# Make a test director
mkdir zendesk_api_test
cd zendesk_api_test

# Create a composer file
vi composer.json
{
  "require": {
    "zendesk/zendesk_api_client_php": "dev-master"
  }
}

# Install depedencies
composer install

# Create a PHP file
vi test.php

<?php

// load Composer
require 'vendor/autoload.php';

use Zendesk\API\HttpClient as ZendeskAPI;

$subdomain = "<subdomain goes here>";
$username  = "<email goes here>";
$token     = "<token goes here>";

$client = new ZendeskAPI($subdomain, $username);
$client->setAuth('basic', ['username' => $username, 'token' => $token]);

$tickets = $client->tickets()->findAll();

print_r($tickets);

?>

php test.php
```

**Comment by aprast** [comment]
ok, thanks.

Will try tomorrow.

**Comment by aprast** [comment]
Ok,

Did what you said on new machine and still nothing.

Attached is my test.php script
and output.txt  I include phpinfo() in the test.php script.

Running on Mac OS X clean install in directory.

Thanks.

Albert


On Fri, Sep 18, 2015 at 9:10 PM, Jason Smale <notifications@github.com>
wrote:

> <?php // load Composer require 'vendor/autoload.php'; use Zendesk\API\HttpClient
> as ZendeskAPI; $subdomain = "<subdomain goes here>"; $username = "<email
> goes here>"; $token = "<token goes here>"; $client = new ZendeskAPI(
> $subdomain, $username); $client->setAuth('basic', ['username' => $username,
> 'token' => $token]); $tickets = $client->tickets()->findAll(); print_r(
> $tickets); ?>

phpinfo()
PHP Version => 5.5.10

System => Darwin AAPKitchen.local 14.5.0 Darwin Kernel Version 14.5.0: Wed Jul 29 02:26:53 PDT 2015; root:xnu-2782.40.9~1/RELEASE_X86_64 x86_64
Build Date => Apr 10 2014 17:49:03
Configure Command =>  './configure'  '--with-mysql=/Applications/MAMP/Library' '--with-gd' '--with-jpeg-dir=/Applications/MAMP/Library' '--with-png-dir=/Applications/MAMP/Library' '--with-zlib' '--with-zlib-dir=/Applications/MAMP/Library' '--with-freetype-dir=/Applications/MAMP/Library' '--prefix=/Applications/MAMP/bin/php/php5.5.10' '--exec-prefix=/Applications/MAMP/bin/php/php5.5.10' '--sysconfdir=/Applications/MAMP/bin/php/php5.5.10/conf' '--with-config-file-path=/Applications/MAMP/bin/php/php5.5.10/conf' '--enable-ftp' '--enable-gd-native-ttf' '--with-bz2=/usr' '--with-ldap' '--with-mysqli=/Applications/MAMP/Library/bin/mysql_config' '--with-t1lib=/Applications/MAMP/Library' '--enable-mbstring=all' '--with-curl=/Applications/MAMP/Library' '--enable-sockets' '--enable-bcmath' '--with-imap=shared,/Applications/MAMP/Library/lib/imap-2007f' '--enable-soap' '--with-kerberos' '--enable-calendar' '--with-pgsql=shared,/Applications/MAMP/Library/pg' '--enable-exif' '--with-libxml-dir=/Applications/MAMP/Library' '--with-gettext=shared,/Applications/MAMP/Library' '--with-xsl=/Applications/MAMP/Library' '--with-pdo-mysql=shared,/Applications/MAMP/Library' '--with-pdo-pgsql=shared,/Applications/MAMP/Library/pg' '--with-mcrypt=shared,/Applications/MAMP/Library' '--with-openssl' '--enable-zip' '--with-iconv=/Applications/MAMP/Library' '--enable-opcache' '--enable-cgi' '--enable-intl' '--with-icu-dir=/Applications/MAMP/Library' '--with-tidy=shared'
Server API => Command Line Interface
Virtual Directory Support => disabled
Configuration File (php.ini) Path => /Applications/MAMP/bin/php/php5.5.10/conf
Loaded Configuration File => /Applications/MAMP/bin/php/php5.5.10/conf/php.ini
Scan this dir for additional .ini files => (none)
Additional .ini files parsed => (none)
PHP API => 20121113
PHP Extension => 20121212
Zend Extension => 220121212
Zend Extension Build => API220121212,NTS
PHP Extension Build => API20121212,NTS
Debug Build => no
Thread Safety => disabled
Zend Signal Handling => disabled
Zend Memory Manager => enabled
Zend Multibyte Support => provided by mbstring
IPv6 Support => enabled
DTrace Support => disabled

Registered PHP Streams => https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, phar, zip
Registered Stream Socket Transports => tcp, udp, unix, udg, ssl, sslv3, sslv2, tls
Registered Stream Filters => zlib.*, bzip2.*, convert.iconv.*, string.rot13, string.toupper, string.tolower, string.strip_tags, convert.*, consumed, dechunk, mcrypt.*, mdecrypt.*

This program makes use of the Zend Scripting Language Engine:
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies


 _______________________________________________________________________


Configuration

bcmath

BCMath support => enabled

Directive => Local Value => Master Value
bcmath.scale => 0 => 0

bz2

BZip2 Support => Enabled
Stream Wrapper support => compress.bzip2://
Stream Filter support => bzip2.decompress, bzip2.compress
BZip2 Version => 1.0.6, 6-Sept-2010

calendar

Calendar support => enabled

Core

PHP Version => 5.5.10

Directive => Local Value => Master Value
allow_url_fopen => On => On
allow_url_include => Off => Off
always_populate_raw_post_data => Off => Off
arg_separator.input => & => &
arg_separator.output => & => &
asp_tags => Off => Off
auto_append_file => no value => no value
auto_globals_jit => On => On
auto_prepend_file => no value => no value
browscap => no value => no value
default_charset => no value => no value
default_mimetype => text/html => text/html
disable_classes => no value => no value
disable_functions => no value => no value
display_errors => Off => Off
display_startup_errors => Off => Off
doc_root => no value => no value
docref_ext => no value => no value
docref_root => no value => no value
enable_dl => On => On
enable_post_data_reading => On => On
error_append_string => no value => no value
error_log => /Applications/MAMP/logs/php_error.log => /Applications/MAMP/logs/php_error.log
error_prepend_string => no value => no value
error_reporting => 32767 => 32767
exit_on_timeout => Off => Off
expose_php => On => On
extension_dir => /Applications/MAMP/bin/php/php5.5.10/lib/php/extensions/no-debug-non-zts-20121212/ => /Applications/MAMP/bin/php/php5.5.10/lib/php/extensions/no-debug-non-zts-20121212/
file_uploads => On => On
highlight.comment => <font style="color: #FF8000">#FF8000</font> => <font style="color: #FF8000">#FF8000</font>
highlight.default => <font style="color: #0000BB">#0000BB</font> => <font style="color: #0000BB">#0000BB</font>
highlight.html => <font style="color: #000000">#000000</font> => <font style="color: #000000">#000000</font>
highlight.keyword => <font style="color: #007700">#007700</font> => <font style="color: #007700">#007700</font>
highlight.string => <font style="color: #DD0000">#DD0000</font> => <font style="color: #DD0000">#DD0000</font>
html_errors => Off => Off
ignore_repeated_errors => Off => Off
ignore_repeated_source => Off => Off
ignore_user_abort => Off => Off
implicit_flush => On => On
include_path => .:/Applications/MAMP/bin/php/php5.5.10/lib/php => .:/Applications/MAMP/bin/php/php5.5.10/lib/php
log_errors => On => On
log_errors_max_len => 1024 => 1024
mail.add_x_header => Off => Off
mail.force_extra_parameters => no value => no value
mail.log => no value => no value
max_execution_time => 0 => 0
max_file_uploads => 20 => 20
max_input_nesting_level => 64 => 64
max_input_time => -1 => -1
max_input_vars => 1000 => 1000
memory_limit => 128M => 128M
open_basedir => no value => no value
output_buffering => 0 => 0
output_handler => no value => no value
post_max_size => 32M => 32M
precision => 12 => 12
realpath_cache_size => 16K => 16K
realpath_cache_ttl => 120 => 120
register_argc_argv => On => On
report_memleaks => On => On
report_zend_debug => Off => Off
request_order => no value => no value
sendmail_from => no value => no value
sendmail_path => /usr/sbin/sendmail -t -i  => /usr/sbin/sendmail -t -i 
serialize_precision => 100 => 100
short_open_tag => On => On
SMTP => localhost => localhost
smtp_port => 25 => 25
sql.safe_mode => Off => Off
sys_temp_dir => no value => no value
track_errors => Off => Off
unserialize_callback_func => no value => no value
upload_max_filesize => 32M => 32M
upload_tmp_dir => /Applications/MAMP/tmp/php => /Applications/MAMP/tmp/php
user_dir => no value => no value
user_ini.cache_ttl => 300 => 300
user_ini.filename => .user.ini => .user.ini
variables_order => EGPCS => EGPCS
xmlrpc_error_number => 0 => 0
xmlrpc_errors => Off => Off
zend.detect_unicode => On => On
zend.enable_gc => On => On
zend.multibyte => Off => Off
zend.script_encoding => no value => no value

ctype

ctype functions => enabled

curl

cURL support => enabled
cURL Information => 7.28.1
Age => 3
Features
AsynchDNS => No
CharConv => No
Debug => No
GSS-Negotiate => No
IDN => Yes
IPv6 => Yes
krb4 => No
Largefile => Yes
libz => Yes
NTLM => Yes
NTLMWB => Yes
SPNEGO => No
SSL => Yes
SSPI => No
TLS-SRP => No
Protocols => dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtsp, smtp, smtps, telnet, tftp
Host => x86_64-apple-darwin10.8.0
SSL Version => OpenSSL/0.9.8}
ZLib Version => 1.2.8

date

date/time support => enabled
"Olson" Timezone Database Version => 2013.9
Timezone Database => internal
Default timezone => Europe/Berlin

Directive => Local Value => Master Value
date.default_latitude => 31.7667 => 31.7667
date.default_longitude => 35.2333 => 35.2333
date.sunrise_zenith => 90.583333 => 90.583333
date.sunset_zenith => 90.583333 => 90.583333
date.timezone => Europe/Berlin => Europe/Berlin

dom

DOM/XML => enabled
DOM/XML API Version => 20031129
libxml Version => 2.8.0
HTML Support => enabled
XPath Support => enabled
XPointer Support => enabled
Schema Support => enabled
RelaxNG Support => enabled

ereg

Regex Library => Bundled library enabled

exif

EXIF Support => enabled
EXIF Version => 1.4 $Id$
Supported EXIF Version => 0220
Supported filetypes => JPEG,TIFF

Directive => Local Value => Master Value
exif.decode_jis_intel => JIS => JIS
exif.decode_jis_motorola => JIS => JIS
exif.decode_unicode_intel => UCS-2LE => UCS-2LE
exif.decode_unicode_motorola => UCS-2BE => UCS-2BE
exif.encode_jis => no value => no value
exif.encode_unicode => ISO-8859-15 => ISO-8859-15

fileinfo

fileinfo support => enabled
version => 1.0.5

filter

Input Validation and Filtering => enabled
Revision => $Id: 4d3899e089e6e45b157975ceef2ac7deb6e9d762 $

Directive => Local Value => Master Value
filter.default => unsafe_raw => unsafe_raw
filter.default_flags => no value => no value

ftp

FTP support => enabled

gd

GD Support => enabled
GD Version => bundled (2.1.0 compatible)
FreeType Support => enabled
FreeType Linkage => with freetype
FreeType Version => 2.4.12
T1Lib Support => enabled
GIF Read Support => enabled
GIF Create Support => enabled
JPEG Support => enabled
libJPEG Version => 8
PNG Support => enabled
libPNG Version => 1.6.6
WBMP Support => enabled
XBM Support => enabled

Directive => Local Value => Master Value
gd.jpeg_ignore_warning => 0 => 0

gettext

GetText Support => enabled

hash

hash support => enabled
Hashing Engines => md2 md4 md5 sha1 sha224 sha256 sha384 sha512 ripemd128 ripemd160 ripemd256 ripemd320 whirlpool tiger128,3 tiger160,3 tiger192,3 tiger128,4 tiger160,4 tiger192,4 snefru snefru256 gost adler32 crc32 crc32b fnv132 fnv164 joaat haval128,3 haval160,3 haval192,3 haval224,3 haval256,3 haval128,4 haval160,4 haval192,4 haval224,4 haval256,4 haval128,5 haval160,5 haval192,5 haval224,5 haval256,5 

iconv

iconv support => enabled
iconv implementation => libiconv
iconv library version => 1.14

Directive => Local Value => Master Value
iconv.input_encoding => ISO-8859-1 => ISO-8859-1
iconv.internal_encoding => ISO-8859-1 => ISO-8859-1
iconv.output_encoding => ISO-8859-1 => ISO-8859-1

imap

IMAP c-Client Version => 2007f
Kerberos Support => enabled

intl

Internationalization support => enabled
version => 1.1.0
ICU version => 52.1
ICU Data version => 52.1

Directive => Local Value => Master Value
intl.default_locale => no value => no value
intl.error_level => 0 => 0
intl.use_exceptions => 0 => 0

json

json support => enabled
json version => 1.2.1

ldap

LDAP Support => enabled
RCS Version => $Id$
Total Links => 0/unlimited
API Version => 3001
Vendor Name => OpenLDAP
Vendor Version => 20411

Directive => Local Value => Master Value
ldap.max_links => Unlimited => Unlimited

libxml

libXML support => active
libXML Compiled Version => 2.8.0
libXML Loaded Version => 20800
libXML streams => enabled

mbstring

Multibyte Support => enabled
Multibyte string engine => libmbfl
HTTP input encoding translation => disabled
libmbfl version => 1.3.2

mbstring extension makes use of "streamable kanji code filter and converter", which is distributed under the GNU Lesser General Public License version 2.1.

Multibyte (japanese) regex support => enabled
Multibyte regex (oniguruma) backtrack check => On
Multibyte regex (oniguruma) version => 5.9.2

Directive => Local Value => Master Value
mbstring.detect_order => no value => no value
mbstring.encoding_translation => Off => Off
mbstring.func_overload => 0 => 0
mbstring.http_input => pass => pass
mbstring.http_output => pass => pass
mbstring.http_output_conv_mimetypes => ^(text/|application/xhtml\+xml) => ^(text/|application/xhtml\+xml)
mbstring.internal_encoding => no value => no value
mbstring.language => neutral => neutral
mbstring.strict_detection => Off => Off
mbstring.substitute_character => no value => no value

mcrypt

mcrypt support => enabled
mcrypt_filter support => enabled
Version => 2.5.8
Api No => 20021217
Supported ciphers => cast-128 gost rijndael-128 twofish arcfour cast-256 loki97 rijndael-192 saferplus wake blowfish-compat des rijndael-256 serpent xtea blowfish enigma rc2 tripledes 
Supported modes => cbc cfb ctr ecb ncfb nofb ofb stream 

Directive => Local Value => Master Value
mcrypt.algorithms_dir => no value => no value
mcrypt.modes_dir => no value => no value

mysql

MySQL Support => enabled
Active Persistent Links => 0
Active Links => 0
Client API version => 5.5.34
MYSQL_MODULE_TYPE => external
MYSQL_SOCKET => /Applications/MAMP/tmp/mysql/mysql.sock
MYSQL_INCLUDE => -I/Applications/MAMP/Library/include
MYSQL_LIBS => -L/Applications/MAMP/Library/lib -lmysqlclient 

Directive => Local Value => Master Value
mysql.allow_local_infile => On => On
mysql.allow_persistent => On => On
mysql.connect_timeout => 60 => 60
mysql.default_host => no value => no value
mysql.default_password => no value => no value
mysql.default_port => no value => no value
mysql.default_socket => /Applications/MAMP/tmp/mysql/mysql.sock => /Applications/MAMP/tmp/mysql/mysql.sock
mysql.default_user => no value => no value
mysql.max_links => Unlimited => Unlimited
mysql.max_persistent => Unlimited => Unlimited
mysql.trace_mode => Off => Off

mysqli

MysqlI Support => enabled
Client API library version => 5.5.34
Active Persistent Links => 0
Inactive Persistent Links => 0
Active Links => 0
Client API header version => 5.5.34
MYSQLI_SOCKET => /Applications/MAMP/tmp/mysql/mysql.sock

Directive => Local Value => Master Value
mysqli.allow_local_infile => On => On
mysqli.allow_persistent => On => On
mysqli.default_host => no value => no value
mysqli.default_port => 3306 => 3306
mysqli.default_pw => no value => no value
mysqli.default_socket => no value => no value
mysqli.default_user => no value => no value
mysqli.max_links => Unlimited => Unlimited
mysqli.max_persistent => Unlimited => Unlimited
mysqli.reconnect => Off => Off

openssl

OpenSSL support => enabled
OpenSSL Library Version => OpenSSL 0.9.8zg 14 July 2015
OpenSSL Header Version => OpenSSL 0.9.8r 8 Feb 2011

pcre

PCRE (Perl Compatible Regular Expressions) Support => enabled
PCRE Library Version => 8.34 2013-12-15

Directive => Local Value => Master Value
pcre.backtrack_limit => 1000000 => 1000000
pcre.recursion_limit => 100000 => 100000

PDO

PDO support => enabled
PDO drivers => sqlite, pgsql, mysql

pdo_mysql

PDO Driver for MySQL => enabled
Client API version => 5.5.34

Directive => Local Value => Master Value
pdo_mysql.default_socket => /Applications/MAMP/tmp/mysql/mysql.sock => /Applications/MAMP/tmp/mysql/mysql.sock

pdo_pgsql

PDO Driver for PostgreSQL => enabled
PostgreSQL(libpq) Version => 8.4.17
Module version => 1.0.2
Revision =>  $Id$ 

pdo_sqlite

PDO Driver for SQLite 3.x => enabled
SQLite Library => 3.7.7.1

pgsql

PostgreSQL Support => enabled
PostgreSQL(libpq) Version => 8.4.17
PostgreSQL(libpq)  => PostgreSQL 8.4.17 on i386-apple-darwin10.8.0, compiled by GCC i686-apple-darwin10-llvm-gcc-4.2 (GCC) 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.1.00), 64-bit
Multibyte character support => enabled
SSL support => enabled
Active Persistent Links => 0
Active Links => 0

Directive => Local Value => Master Value
pgsql.allow_persistent => On => On
pgsql.auto_reset_persistent => Off => Off
pgsql.ignore_notice => Off => Off
pgsql.log_notice => Off => Off
pgsql.max_links => Unlimited => Unlimited
pgsql.max_persistent => Unlimited => Unlimited

Phar

Phar: PHP Archive support => enabled
Phar EXT version => 2.0.2
Phar API version => 1.1.1
SVN revision => $Id: a5488937a80433d434af6753d3db8517feaf91df $
Phar-based phar archives => enabled
Tar-based phar archives => enabled
ZIP-based phar archives => enabled
gzip compression => enabled
bzip2 compression => enabled
Native OpenSSL support => enabled


Phar based on pear/PHP_Archive, original concept by Davey Shafik.
Phar fully realized by Gregory Beaver and Marcus Boerger.
Portions of tar implementation Copyright (c) 2003-2009 Tim Kientzle.
Directive => Local Value => Master Value
phar.cache_list => no value => no value
phar.readonly => On => On
phar.require_hash => On => On

posix

Revision => $Id: 1dfa9997ed76804e53c91e0ce862f3707617b6ed $

Reflection

Reflection => enabled
Version => $Id: 31d836a7ac92a37b5c580836d91ad4736fe2f376 $

session

Session Support => enabled
Registered save handlers => files user 
Registered serializer handlers => php_serialize php php_binary 

Directive => Local Value => Master Value
session.auto_start => Off => Off
session.cache_expire => 180 => 180
session.cache_limiter => nocache => nocache
session.cookie_domain => no value => no value
session.cookie_httponly => Off => Off
session.cookie_lifetime => 0 => 0
session.cookie_path => / => /
session.cookie_secure => Off => Off
session.entropy_file => no value => no value
session.entropy_length => 0 => 0
session.gc_divisor => 100 => 100
session.gc_maxlifetime => 1440 => 1440
session.gc_probability => 1 => 1
session.hash_bits_per_character => 4 => 4
session.hash_function => 0 => 0
session.name => PHPSESSID => PHPSESSID
session.referer_check => no value => no value
session.save_handler => files => files
session.save_path => /Applications/MAMP/tmp/php => /Applications/MAMP/tmp/php
session.serialize_handler => php => php
session.upload_progress.cleanup => On => On
session.upload_progress.enabled => On => On
session.upload_progress.freq => 1% => 1%
session.upload_progress.min_freq => 1 => 1
session.upload_progress.name => PHP_SESSION_UPLOAD_PROGRESS => PHP_SESSION_UPLOAD_PROGRESS
session.upload_progress.prefix => upload_progress_ => upload_progress_
session.use_cookies => On => On
session.use_only_cookies => On => On
session.use_strict_mode => Off => Off
session.use_trans_sid => 0 => 0

SimpleXML

Simplexml support => enabled
Revision => $Id: 363ea1fc44b0b7775e2ec3ce8456530fec6d1660 $
Schema support => enabled

soap

Soap Client => enabled
Soap Server => enabled

Directive => Local Value => Master Value
soap.wsdl_cache => 1 => 1
soap.wsdl_cache_dir => /tmp => /tmp
soap.wsdl_cache_enabled => 1 => 1
soap.wsdl_cache_limit => 5 => 5
soap.wsdl_cache_ttl => 86400 => 86400

sockets

Sockets Support => enabled

SPL

SPL support => enabled
Interfaces => Countable, OuterIterator, RecursiveIterator, SeekableIterator, SplObserver, SplSubject
Classes => AppendIterator, ArrayIterator, ArrayObject, BadFunctionCallException, BadMethodCallException, CachingIterator, CallbackFilterIterator, DirectoryIterator, DomainException, EmptyIterator, FilesystemIterator, FilterIterator, GlobIterator, InfiniteIterator, InvalidArgumentException, IteratorIterator, LengthException, LimitIterator, LogicException, MultipleIterator, NoRewindIterator, OutOfBoundsException, OutOfRangeException, OverflowException, ParentIterator, RangeException, RecursiveArrayIterator, RecursiveCachingIterator, RecursiveCallbackFilterIterator, RecursiveDirectoryIterator, RecursiveFilterIterator, RecursiveIteratorIterator, RecursiveRegexIterator, RecursiveTreeIterator, RegexIterator, RuntimeException, SplDoublyLinkedList, SplFileInfo, SplFileObject, SplFixedArray, SplHeap, SplMinHeap, SplMaxHeap, SplObjectStorage, SplPriorityQueue, SplQueue, SplStack, SplTempFileObject, UnderflowException, UnexpectedValueException

sqlite3

SQLite3 support => enabled
SQLite3 module version => 0.7-dev
SQLite Library => 3.7.7.1

Directive => Local Value => Master Value
sqlite3.extension_dir => no value => no value

standard

Dynamic Library Support => enabled
Path to sendmail => /usr/sbin/sendmail -t -i 

Directive => Local Value => Master Value
assert.active => 1 => 1
assert.bail => 0 => 0
assert.callback => no value => no value
assert.quiet_eval => 0 => 0
assert.warning => 1 => 1
auto_detect_line_endings => 0 => 0
default_socket_timeout => 60 => 60
from => no value => no value
url_rewriter.tags => a=href,area=href,frame=src,input=src,form=,fieldset= => a=href,area=href,frame=src,input=src,form=,fieldset=
user_agent => no value => no value

tokenizer

Tokenizer Support => enabled

xml

XML Support => active
XML Namespace Support => active
libxml2 Version => 2.8.0

xmlreader

XMLReader => enabled

xmlwriter

XMLWriter => enabled

xsl

XSL => enabled
libxslt Version => 1.1.28
libxslt compiled against libxml Version => 2.8.0
EXSLT => enabled
libexslt Version => 1.1.28

yaz

YAZ Support => enabled
PHP/YAZ Version => 1.1.5
YAZ Version => 4.0.1
Compiled with YAZ version => 4.0.1

zip

Zip => enabled
Extension Version => $Id: 05dd1ecc211075107543b0ef8cee488dd229fccf $
Zip version => 1.11.0
Libzip version => 0.10.1

zlib

ZLib Support => enabled
Stream Wrapper => compress.zlib://
Stream Filter => zlib.inflate, zlib.deflate
Compiled Version => 1.2.8
Linked Version => 1.2.8

Directive => Local Value => Master Value
zlib.output_compression => Off => Off
zlib.output_compression_level => -1 => -1
zlib.output_handler => no value => no value

Additional Modules

Module Name

Environment

Variable => Value
TERM_PROGRAM => iTerm.app
SHELL => /bin/bash
TERM => xterm
TMPDIR => /var/folders/gf/rv1kmfjx0b5d2fgd89zbf0r80000gn/T/
Apple_PubSub_Socket_Render => /private/tmp/com.apple.launchd.Vlfkj3Tiju/Render
USER => aprast
SSH_AUTH_SOCK => /private/tmp/com.apple.launchd.wGPntjcuYi/Listeners
__CF_USER_TEXT_ENCODING => 0x1F5:0x0:0x0
PATH => /Applications/MAMP/bin/php/php5.5.10/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
PWD => /Users/aprast/ZD
LANG => en_US.UTF-8
ITERM_PROFILE => Default
XPC_FLAGS => 0x0
XPC_SERVICE_NAME => 0
SHLVL => 1
HOME => /Users/aprast
ITERM_SESSION_ID => w0t0p0
LOGNAME => aprast
_ => /Applications/MAMP/bin/php/php5.5.10/bin/php
OLDPWD => /Users/aprast

PHP Variables

Variable => Value
_SERVER["TERM_PROGRAM"] => iTerm.app
_SERVER["SHELL"] => /bin/bash
_SERVER["TERM"] => xterm
_SERVER["TMPDIR"] => /var/folders/gf/rv1kmfjx0b5d2fgd89zbf0r80000gn/T/
_SERVER["Apple_PubSub_Socket_Render"] => /private/tmp/com.apple.launchd.Vlfkj3Tiju/Render
_SERVER["USER"] => aprast
_SERVER["SSH_AUTH_SOCK"] => /private/tmp/com.apple.launchd.wGPntjcuYi/Listeners
_SERVER["__CF_USER_TEXT_ENCODING"] => 0x1F5:0x0:0x0
_SERVER["PATH"] => /Applications/MAMP/bin/php/php5.5.10/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
_SERVER["PWD"] => /Users/aprast/ZD
_SERVER["LANG"] => en_US.UTF-8
_SERVER["ITERM_PROFILE"] => Default
_SERVER["XPC_FLAGS"] => 0x0
_SERVER["XPC_SERVICE_NAME"] => 0
_SERVER["SHLVL"] => 1
_SERVER["HOME"] => /Users/aprast
_SERVER["ITERM_SESSION_ID"] => w0t0p0
_SERVER["LOGNAME"] => aprast
_SERVER["_"] => /Applications/MAMP/bin/php/php5.5.10/bin/php
_SERVER["OLDPWD"] => /Users/aprast
_SERVER["PHP_SELF"] => test.php
_SERVER["SCRIPT_NAME"] => test.php
_SERVER["SCRIPT_FILENAME"] => test.php
_SERVER["PATH_TRANSLATED"] => test.php
_SERVER["DOCUMENT_ROOT"] => 
_SERVER["REQUEST_TIME_FLOAT"] => 1442626652.3
_SERVER["REQUEST_TIME"] => 1442626652
_SERVER["argv"] => Array
(
    [0] => test.php
)

_SERVER["argc"] => 1
_ENV["TERM_PROGRAM"] => iTerm.app
_ENV["SHELL"] => /bin/bash
_ENV["TERM"] => xterm
_ENV["TMPDIR"] => /var/folders/gf/rv1kmfjx0b5d2fgd89zbf0r80000gn/T/
_ENV["Apple_PubSub_Socket_Render"] => /private/tmp/com.apple.launchd.Vlfkj3Tiju/Render
_ENV["USER"] => aprast
_ENV["SSH_AUTH_SOCK"] => /private/tmp/com.apple.launchd.wGPntjcuYi/Listeners
_ENV["__CF_USER_TEXT_ENCODING"] => 0x1F5:0x0:0x0
_ENV["PATH"] => /Applications/MAMP/bin/php/php5.5.10/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
_ENV["PWD"] => /Users/aprast/ZD
_ENV["LANG"] => en_US.UTF-8
_ENV["ITERM_PROFILE"] => Default
_ENV["XPC_FLAGS"] => 0x0
_ENV["XPC_SERVICE_NAME"] => 0
_ENV["SHLVL"] => 1
_ENV["HOME"] => /Users/aprast
_ENV["ITERM_SESSION_ID"] => w0t0p0
_ENV["LOGNAME"] => aprast
_ENV["_"] => /Applications/MAMP/bin/php/php5.5.10/bin/php
_ENV["OLDPWD"] => /Users/aprast

PHP License
This program is free software; you can redistribute it and/or modify
it under the terms of the PHP License as published by the PHP Group
and included in the distribution in the file:  LICENSE

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

If you did not receive a copy of the PHP license, or have any
questions about PHP licensing, please contact license@php.net.

**Comment by jwswj** [comment]
You can't attach things to Github via email. You'll need to host the script in gist or somewhere.

We've got several people running this client on Mac's using php 5.5 and 5.6, so it might be some weird configuration issues. You'll need to provide more information than just you're PHP version and script. Please provide exactly the commands you are running and possibly a screen recording.

Please also try running the commands I outlined above on a different machine (either mac or linux) to make sure it's not an individual machine.

**Comment by vishalnct** [comment]
I also face this error Fatal error: Call to a member function getReasonPhrase() on a non-object.
Please provide me best solution.

**Comment by jwswj** [comment]
@

**Comment by vishalnct** [comment]
I used this demo https://github.com/zendesk/zendesk_api_client_php/blob/master/samples/getTickets.php but when i run error is occur.
My link http://easyquote.se/zendesk_api/samples/getTickets.php

**Comment by jwswj** [comment]
@vishalnct we're going to need more information. How exactly did you install composer? what permissions are setup on the composer executable? Can you run composer on your machine?

https://laracasts.com/discuss/channels/general-discussion/call-to-a-member-function-make-on-null would indicate that this might be a composer issue (maybe permissions relate?)

**Comment by vishalnct** [comment]
yes i run composer on my machine by command prompt.

**Comment by miogalang** [comment]
Hi @vishalnct,

From the error you are describing it seems the PHP file is not finding the autoload.php which composer generates on your machine. Please check the path to the file is correct. Please note that the path (in this case `../vendor/autoload.php`) is relative to the PHP file you are executing.

**Comment by jwswj** [comment]
I'm closing this as we have tried to re-create the issue on multiple machines and haven't been able to. We'd like to get to the bottom of it, but don't believe it's an issue with the API client, rather the configuration setup on individual machines. The comment I made earlier in this thread (https://github.com/zendesk/zendesk_api_client_php/issues/194#issuecomment-141605856) shows the commands which can be run on a clean ubuntu install without any problems.
